zerver/tests/test_upload.py

```diff
@@ -538,6 +540,19 @@ def test_file_download_authorization_public(self) -> None:
             self.assertEqual(b"zulip!", data)
             self.logout()
 
+    def test_serve_local(self) -> None:
+        with self.settings(SENDFILE_BACKEND='sendfile.backends.nginx'):
+            _get_sendfile.clear()  # To clearout cached version of backend from djangosendfile
+            filename = os.path.join(settings.LOCAL_UPLOADS_DIR, 'files', 'á.jpg')
+            basedir = os.path.dirname(filename)
+            if not os.path.exists(basedir):
+                os.makedirs(basedir)
+            open(filename, 'w').close()
```

Comment:
What's the point of this -- to create an empty file?

Reply:
yup, basically to have a file which could be used to test out django-sendfile.
django-sendfile checks if the file exists or not and we cannot stub that code for this test to work.

Reply:
OK; I'd recommend having the code actually upload a file (the usual Zulip way like many other tests in this file do) instead, and the test should perhaps include checking that it rejects access if the user doesn't have access to the file.